src/lib.ts

```diff
@@ -1221,12 +1220,10 @@ export function setCombatFlags(
   return visitUrl(
     `account.php?${
       ([
-        ...flat(
-          flags.map(({ flag, value }) => [
-            `actions[]=flag_${flag}`,
-            `flag_${flag}=${Number(value)}`,
-          ]),
-        ),
+        ...flags.flatMap(({ flag, value }) => [
```

Comment:
now that I read this in more detail this looks like another `[...x]` that can be simplified to `x`.